cmake/toolchain/xtools/Kconfig

```diff
@@ -0,0 +1,6 @@
+# Copyright © 2022 Keith Packard
+# SPDX-License-Identifier: Apache-2.0
+
+config TOOLCHAIN_XTOOLS_SUPPORTS_THREAD_LOCAL_STORAGE
+	def_bool y
+	select TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
```

Comment:
Is this intentionally promptless? (shouldn't users be able to override this?)

If not:

```kconfig
config TOOLCHAIN_XTOOLS_SUPPORTS_THREAD_LOCAL_STORAGE
	bool "Crosstool-ng toolchain supports TLS"
	default y
	select TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
```

Reply:
no, cause if you follow the guide you'll be building cross-tools according to how we build Zephyr SDK:
https://docs.zephyrproject.org/latest/develop/toolchains/crosstool_ng.html

and then TLS will be supported.

btw. @stephanosio the guide still refers to use `./go.sh <arch>` but that has been removed in the sdk-ng repo, so what is the right procedure now ?

Reply:
Looks like the whole guide needs to be reworked. crosstool-ng is a standalone (independent from both Zephyr and Zephyr SDK) component and the toolchain build configuration can be whatever user chooses.

I am not sure if there is any value in keeping the `xtools` toolchain variant because you may as well be using the `cross-compile`.

Reply:
I will post a separate PR removing the `xtools` toolchain variant since it does nothing.

Reply:
That seems quite reasonable -- I've been testing these patches with crosstool-ng toolchains (both arm-none-eabi and arm-zephyr-eabi) as well as the native Debian arm-none-eabi toolchain.

Reply:
> I am not sure if there is any value in keeping the xtools toolchain variant because you may as well be using the cross-compile.

@stephanosio and @keith-packard the biggest value I see is that if using the Zephyr crosstool setup, you'll get the `zephyr` triplet and the arch selection of compiler.
With `cross-compile` you must adjust `CROSS_COMPILE` variable when building for a new architecture.

The `xtools` CMake files also serves as a nice template for those creating custom toolchains for many archs, as they may easily see an example of how automatic selection of arch a specific toolchain can be done.

So `xtools` is a conveniance and template, but everything can be achieved using `cross-compile`, just as `ZEPHYR_TOOLCHAIN_VARIANT=zephyr` is a convenience.

We could request users to point to their Zephyr SDK using `export CROSS_COMPILE=/opt/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-`, but we don't, cause we want an easy way to use Zephyr SDK.

Reply:
"Zephyr crosstool" is not really a thing. If you are using Zephyr toolchains from the Zephyr SDK, then you would be using the `zephyr` toolchain variant. If not, you will be using the `cross-compile` toolchain variant.

crosstool-ng can build any toolchain from `aarch64-none-elf` to `aarch64-linux-gnu` -- the `xtools` toolchain variant, as it is implemented and stated in the docs, does not really make much sense.